code_generator/GdkCodeGenerator/Templates/UnityReaderWriterGenerator.tt

```diff
@@ -23,44 +23,48 @@ namespace <#= qualifiedNamespace #>
         [ComponentId(<#= unityComponentDefinition.Id #>)]
         internal class ReaderWriterCreator : IReaderWriterCreator
         {
-            public IReaderInternal CreateReaderWriter(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
+            public IReaderWriterInternal CreateReaderWriter(Entity entity, EntityManager entityManager, ILogDispatcher logDispatcher)
             {
                 return new ReaderWriterImpl(entity, entityManager, logDispatcher);
             }
         }
 
         [ReaderInterface]
         [ComponentId(<#= unityComponentDefinition.Id #>)]
-        public interface Reader
+        public interface Reader : IReader<<#= componentDetails.TypeName #><# if(fieldDetailsList.Count > 0){ #>, <#= componentDetails.TypeName #>.Update<# } #>>
         {
         }
 
-        public class ReaderWriterImpl : Reader, IReaderInternal
+        [WriterInterface]
```

Comment:
Isn't this tag redundant? Given the interface name?

Reply:
The attribute marks it as a component-specific writer interface and is picked up by the SpatialOSLibrary, which identifies required readers and writers by looking for these tags via reflection.

Reply:
You can check whether the type implements the interface?

Reply:
That could work too, but could get a bit unintuitive with Writers being Readers as well, and currently there is no general "Writer" superinterface. The decision was basically having an empty tag interface with slightly confusing details vs attribute tags, and the version with attribute tags was already written at that point :)

Reply:
It feels like a code smell to be - but we can revisit this after the MVP is in